proposals/devcontainer-collections.md

```diff
@@ -0,0 +1,114 @@
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| definitions | array | The list of definitions contained in this collection. Same information as the metadata file for the definition.|
+
+Creating the collections file would make some changes over the metadata of the features/definitions where they are generated.
+
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+
+The same way creating the definitions/feature file would also be part of the release process for it so that the following metadata can be added:
+- Version.
+- Checksum.
+- Signatures.
+- Date.
+
+## Definitions/features (devcontainer-feature.json/devcontainer-definition.json)
+
+### Common Properties
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The ids should be unique in the contest of the repository where they exist. |
+| displayName | string | Name of the feature/definition |
+| shortDescription | string | Optional one sentence description |
+| description | string | Description of the feature/definition |
+| metadata | any | Freeform data added by users for their own purposes. |
+| repositoryPath | string | Path to the repository and folder that contains the root of the code for the feature/definition. |
+| keywords | array | List of keywords relevant to a user that would search for this definition/feature. |
+| version | string | Versioning information for the feature/definition. |
+| files | array | List of links for the download files with checksums and optional signature information. |
+
+### Definitions
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| type | string | singleContainer/orchestrated |
+| options | array | List of user selected options at the moment the user selects them in a client application. |
+| variants | array | Variations of the generated definition that contain different information. (e.g. with example code or data.) |
+
+Variants:
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| name | string | Name of the variant |
+| description | string | Short description of the variant |
+
+### Features
+
+TODO: Add properties related to references/dependencies.
+
+# IDs
+
+TODO
+
+# Search
+
+Search would be implemented by client applications but the definitions themselves should include enough information for this to happen. Besides the typed properties defined above keywords could be:
+
+- Purpose. WebApp, WebService, IDE extension, etc.
+- Platform. Node, Ruby on Rails, .Net, etc.
+- Language. TS, JS, C#, Java, etc.
+- Other Apps. Databases, simulators, etc.
+- Dependencies. Bare metal, Azure, AWS, etc.
+- Organization. 
```

Comment:
Additional data that we might want to represent as separate properties (not as keywords) because we are using them programmatically (currently parsed from the definition's README.md):
- A flag for 'core definitions' to show only these when we don't have any other way to filter down the list of definitions in the UI. (Not sure how to represent that when we accept anyones definitions, maybe this should be a list on the products' side.)
- Lists of platforms/languages to filter down the list of definitions when we know some of the file types used in the repository. (This can be done as keywords if that's preferred.)
- A flag indicating compatibility with Codespaces: To be product-neutral, we might want to have a list of known compatible products and a list of known incompatible products, so each product can then decide whether to include a definition that does not list the product at all.